### PR TITLE
Feat/be/user dashboard conditional logic

### PR DESCRIPTION
**Pull request by aloraalee**
## Type of Change
- [x] feature ⛲
- [x] testing 🧪

## Description
This PR adds logic to the dashboard to only render job applications, contacts, and companies to the dashboard if they were created in the last seven days. This is done with a method in the dashboard model, and it is tested as well. 

## Motivation and Context
Filtering the data on the backend with Active Record helps to maintain efficiency by only sending data from the last seven days to the dashboard. The reason for the seven-day limit is to incentivize the user to add new job applications, contacts or companies if they have not already done so. 

## Added Test?
- [x] Yes 🫡
 Model Specs
- [x] All previous tests still pass 🥳

## Checklist:
- [x] My code follows the code style of this project.

### CODE REVIEW

File: app/controllers/api/v1/dashboards_controller.rb [hunk 1]
```diff
@@ -4,6 +4,7 @@ class Api::V1::DashboardsController < ApplicationController
   def show
     user = current_user
     authorize user
-    render json: DashboardSerializer.new(user), status: :ok
+    weekly_summary = Dashboard.filter_weekly_summary(user)
```
Comment (TDManning) [inline app/controllers/api/v1/dashboards_controller.rb:7]: Good catch removing the render here and having it render only once. Refactoring into a weekly_summary works well!

File: spec/models/dashboard_spec.rb
Comment (TDManning) [inline spec/models/dashboard_spec.rb:1]: You could also test that the job_applications are being filtered

**Review by TDManning — APPROVED** [review]
This all looks great. See my comment about missing a check that job_applications are also being filtered, if they exist.

File: app/controllers/api/v1/dashboards_controller.rb (hunk 1 above)
Comment (SmilodonP) [reply app/controllers/api/v1/dashboards_controller.rb:7]: 👆

File: app/models/dashboard.rb
Comment (SmilodonP) [inline app/models/dashboard.rb:1]: Well written and concise business logic.

File: spec/models/dashboard_spec.rb
Comment (SmilodonP) [reply spec/models/dashboard_spec.rb:1]: This would definitely make the testing significantly more robust. There should be more than just the one expectation/assertion.

**Review by SmilodonP — COMMENTED** [review]
I think that the testing needs to be expanded a bit before this gets merged. The code itself itself looks very solid though.

File: spec/models/dashboard_spec.rb
Comment (aloraalee) [reply spec/models/dashboard_spec.rb:1]: Great I added job_applications!

Comment (aloraalee) [reply spec/models/dashboard_spec.rb:1]: Edge cases and more robust testing has been added!

Comment (MiTOBrien) [inline spec/models/dashboard_spec.rb:1]: Nice, clean test coverage.
